### PR TITLE
DR-457 compression overview

### PR DESCRIPTION
**Pull request by daveatweaviate**
[issue](https://semi-technology.atlassian.net/browse/DR-457)
[staging](https://dr-457-compression-overview--tangerine-buttercream-20c32f.netlify.app/developers/weaviate/starter-guides/compression)

### CODE REVIEW

**Comment by daveatweaviate** [comment]
@sebawita would you have a look at this draft when you get a moment? Thanks!

File: developers/weaviate/starter-guides/compression.mdx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+The over-fetching technique, also call rescoring, means you get the benefits of compression without losing the precision of an uncompressed vector search.
+
+#### Query speed
+
+Compressed vectors are significantly smaller than uncompressed vectors. It is much faster to search a compressed data set than an uncompressed one. After Weaviate creates a candidate list, there is a small time cost to search the uncompressed candidate vectors. This cost is small compared to the cost of searching a large, uncompressed data set, and the improved recall justifies the time spent.
+
+Each compression method has its own characteristics with regard to speed.
+
+- PQ indexes have response rates approach the response rates of uncompressed indexes when recall reaches 97 percent and higher. At those levels of recall, the [speed profile](/blog/pq-rescoring#qps-vs-recall-experiments) for PQ compressed indexes matches the profile for uncompressed indexes. Both indexes handle about 200 queries per second (QPS).
```
Comment (sebawita) [inline developers/weaviate/starter-guides/compression.mdx:87]: > Both indexes handle about 200 queries per second (QPS).

The 200 QPS depends on the data and server spec.
We should remove this sentence.

Comment (sebawita) [reply developers/weaviate/starter-guides/compression.mdx:87]: We can include benchmarks later. But these will have specific params

File: developers/weaviate/starter-guides/compression.mdx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+- SQ significantly improves search speeds. SQ is new in v1.26. It is faster than PQ, perhaps 3 to 4 times as fast as searching uncompressed data. SQ has a higher dimensional resolution than BQ that helps recall. Look for an upcoming blog post that discusses the tradeoffs with SQ compression.
+
+SQ and BQ both have optional vector caches. Use these configurable caches to load frequently used, uncompressed vectors into memory to improve overall search times.
+
+#### Import speed
+
+Importing and compressing vectors takes slightly longer than just importing them, but this is a one time cost. In contrast, loading a compressed index into memory is faster since there is less data to load.
```
Comment (sebawita) [inline developers/weaviate/starter-guides/compression.mdx:97]: > Importing and compressing vectors takes slightly longer than just importing them,

Maybe we could say: "than importing uncompressed vectors" 🤔

File: developers/weaviate/starter-guides/compression.mdx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+Starting in v1.22, Weaviate has an optional, [asynchronous indexing](/developers/weaviate/config-refs/schema/vector-index#asynchronous-indexing) feature which improves the import process. When asynchronous indexing is enabled, indexing takes place as a background process. You can begin other tasks without having to wait for the entire import to be indexed. This means you are able to begin other tasks sooner.
```
Comment (sebawita) [inline developers/weaviate/starter-guides/compression.mdx:99]: Hmmm... I am not sure if this is helpful in the context of compression. 🤔

File: developers/weaviate/starter-guides/compression.mdx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+If you have async indexing and [AutoPQ enabled](/developers/weaviate/configuration/compression/pq-compression#configure-autopq), PQ compression can be started anytime. If not, you should only enable PQ after you have imported enough objects to [train the algorithm](/developers/weaviate/configuration/compression/pq-compression#manually-configure-pq).
+
+BQ doesn't require a training step, however it can only be enabled when you create your collection. BQ cannot be enabled after you start to add data to the collection.
+
+## Recommendations
+
+Most applications benefit from compression.
+
+- The cost savings are significant. In [Weaviate Cloud](https://weaviate.io/pricing), for example, uncompressed collections can be 5 to 8 times as expensive as compressed collections.
```
Comment (sebawita) [inline developers/weaviate/starter-guides/compression.mdx:115]: Let's focus this sentence on the positive side and say that compressed collections can yield xyz savings! (not necessarily in these words)

File: developers/weaviate/starter-guides/compression.mdx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+- If you have specialized needs and a very large data set, consider PQ compression. PQ compression is very configurable, but it requires more expertise to tune well than SQ or BQ.
+- Most users with medium to large data sets should consider SQ compression. SQ compressed vectors are one quarter the size of uncompressed vectors. Searches with SQ are several times faster than searches with uncompressed vectors. Recall is similar to uncompressed data.
+
+Finally, for collections that are small, but that are expected to grow, consider a dynamic index. Configure the collection to use BQ compression while the index is flat and SQ compression when the collection grow large enough to move from a flat index to an HNSW index.
+
+## Related pages
+
+For more information on compression, see these pages.
+
+### Documentation pages
```
Comment (sebawita) [inline developers/weaviate/starter-guides/compression.mdx:126]: Let's add something here to say that you can go to these pages to learn how to configure/enable these compressions

File: developers/weaviate/starter-guides/compression.mdx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+Compressed indexes use less RAM when they are loaded in memory, however they also use more disk space than uncompressed vectors. Weaviate stores the uncompressed vector and the compressed vector index. This means increased disk storage. However, since the cost of RAM is orders of magnitude higher than the cost of disk, the overall cost to use a compressed index is much lower than the cost of using an uncompressed index.
+
+The cost savings are obvious with in memory indexes such as HNSW. The greater the RAM reduction, the smaller the costs.
+
+- Uncompressed data size is roughly 7x greater than PQ compressed data.
+- Uncompressed data size is roughly 4x greater than SQ compressed data.
+- Uncompressed data size is roughly 32x greater than BQ compressed data.
+
+For additional details, refer to the [blog posts](#blog-posts).
```
Comment (sebawita) [inline developers/weaviate/starter-guides/compression.mdx:67]: hmmm... I am not sure if this adds much at this point.
Maybe this sentence could be added at the start of the blog post segment 🤔

**Review by sebawita — CHANGES_REQUESTED** [review]

**Review by daveatweaviate — COMMENTED** [review]
@sebawita thanks for your feedback this is ready for another pass

File: developers/weaviate/starter-guides/compression.mdx [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+BQ doesn't require a training step, however it can only be enabled when you create your collection. BQ cannot be enabled after you start to add data to the collection.
+
+## Recommendations
+
+Most applications benefit from compression.
+
+- The cost savings are significant. In [Weaviate Cloud](https://weaviate.io/pricing), for example, compressed collections can be more than 80% cheaper than uncompressed collections.
+- If you have a small collection that uses a flat index, consider a BQ index. The BQ index is 32 times smaller and much, much faster than the uncompressed equivalent.
+- If you have specialized needs and a very large data set, consider PQ compression. PQ compression is very configurable, but it requires more expertise to tune well than SQ or BQ.
+- Most users with medium to large data sets should consider SQ compression. SQ compressed vectors are one quarter the size of uncompressed vectors. Searches with SQ are several times faster than searches with uncompressed vectors. Recall is similar to uncompressed data.
```
Comment (trengrj) [inline developers/weaviate/starter-guides/compression.mdx:117]: Searches with SQ are several times faster than searches with uncompressed vectors -> Searches with SQ are faster than searches with uncompressed vectors

Comment (daveatweaviate) [reply developers/weaviate/starter-guides/compression.mdx]: Updated, thanks!

**Review by daveatweaviate — COMMENTED** [review]
@trengrj thanks for your feedback 

I've updated the copy here and sent a note to Sebastian in the channel about the other comments.

**Pull request merged**